packages/services/src/providers/__tests__/rpc-spot-integration.test.ts

```diff
@@ -1162,4 +1162,177 @@ describe("RpcSpotProvider - Integration Tests (Real Blockchain)", () => {
     expect(v2Swap?.protocol).toBe("aerodrome");
     console.log(`✓ Combined filter test passed`);
   }, 30000);
+
+  // ===========================================================================
+  // TOKEN SYMBOL RETRIEVAL TESTS
+  // These tests verify on-chain symbol() retrieval works correctly
+  // Used as fallback when price providers return contract address as symbol
+  // ===========================================================================
+
+  // Test fixture: PONKE token on Base
+  // CoinGecko returns the contract address as symbol for this token
+  // On-chain symbol() call returns "PONKE"
+  const PONKE_TOKEN = {
+    address: "0x4a0c64af541439898448659aedcec8e8e819fc53",
+    expectedSymbol: "PONKE",
+    expectedDecimals: 18,
+    chain: "base" as const,
+    description:
+      "Token where CoinGecko returns address as symbol (discovered in production)",
+  };
+
+  // Known tokens with proper symbols for comparison
+  const KNOWN_TOKENS = {
+    USDC: {
+      address: "0x833589fcd6edb6e08f4c7c32d4f71b54bda02913",
+      expectedSymbol: "USDC",
+      expectedDecimals: 6,
+      chain: "base" as const,
+    },
+    WETH: {
+      address: "0x4200000000000000000000000000000000000006",
+      expectedSymbol: "WETH",
+      expectedDecimals: 18,
+      chain: "base" as const,
+    },
+    AERO: {
+      address: "0x940181a94a35a4569e4529a3cdfb74e38fd98631",
+      expectedSymbol: "AERO",
+      expectedDecimals: 18,
+      chain: "base" as const,
+    },
+  };
+
+  test("should retrieve PONKE token symbol from on-chain (CoinGecko returns address)", async () => {
+    if (!process.env.ALCHEMY_API_KEY) {
```

Comment:
should we remove this and just throw an error so that tests dont silently fail?

and also remove all `console.log`